src/Caching/Memory/test/MemoryCacheSetAndRemoveTests.cs

```diff
@@ -553,7 +553,7 @@ public void AddAndReplaceEntries_AreThreadSafe()
                     var entrySize = random.Next(0, 5);
                     cache.Set(random.Next(0, 10), entrySize, new MemoryCacheEntryOptions { Size = entrySize });
                 }
-            }, cts.Token);
```

Comment:
We don't need to pass the token since we check via cts.IsCancellationRequested inside the tasks. Alternatively we can catch the TaskCancelledException from WaitAll